students/stephen/mailroom/mailroom_fp.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python
+
+"""
+Simple Mailroom Program for a nonprofit
+"""
+
+# The script should prompt the user (you) to choose from a menu of 3 actions:
+# “Send a Thank You”, “Create a Report” or “quit”)
+
+import os, sys
+import json
+
+# Create a data object to store donor information
+# donors = {
+#     'Jeff Bezos': [50, 100, 56],
+#     'Mark Zuckerberg': [45, 105, 349, 101, 78],
+#     'Paul Allen': [10, 50, 400, 400],
+#     'Elon Musk': [16, 68, 170, 425],
+#     'Richard Branson': [25, 90, 124, 300]
+# }
+
+class Donor:
+    """
+    Handles the donor
+    """
+    def __init__(self, name, *donations):
+        self._name = name
+        self._donations = list(donations)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @donations.setter
+    def donations(self, lst):
+        self._donations = lst
+
+    def add_donation(self, donation):
+        self._donations.append(donation)
+
+    @property
+    def total_donations(self):
+        return sum(self._donations)
+        # return reduce(lambda a, x: a+x, self._donations, 0)
+        # s = 0
+        # for d in donations:
+        #     s += d
+        # return s
+
+    @property
+    def count_donations(self):
+        return len(self._donations)
+
+    @property
+    def average_donation(self):
+        return self.total_donations / self.count_donations
+
+    @property
+    def letter(self):
+        """
+        Takes the donor iterable and returns a string that is a formatted thank you note
+        with the donor's name and last donation amount.
+        """
+        return "Dear {:s},\n\nWe greatly appreciate your generous donation of ${:,.2f}.\n\nThank you,\nThe Team".format(self._name, self._donations[-1])
+
+    @property
+    def filename(self):
+        'Return a txt file name based on the donor name and using underscores instead of spaces'
+        return self._name.replace(' ', '_') + '.txt'
+
+    # allow other items to call the class
+    def __repr__(self):
+        return "{}: {}".format(self._name, self._donations)  
+
+class DonorDB:
+    """
+    Handles collection of donors
+    """
+    def __init__(self):
+        self._donors = {}
+
+    def add_donor(self, donor):
+        if donor.name.lower() in self._donors:
+            self._donors[donor.name.lower()].add_donation(*(donor.donations))
```

Comment:
something's odd -- shouldn't the donor already have the donations -- would this add it twice ??